### PR TITLE
added spark 2.3.0 support

### PR DESCRIPTION
**Pull request by profes**
This PR adds spark 2.3.0 support

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,6 +24,8 @@ matrix:
       env: PYSPARK_PYTHON=python3 SPARK_VERSION=2.1.2 SPARK_BUILD="spark-${SPARK_VERSION}-bin-hadoop2.7"
     - jdk: oraclejdk8
       env: PYSPARK_PYTHON=python3 SPARK_VERSION=2.2.1 SPARK_BUILD="spark-${SPARK_VERSION}-bin-hadoop2.7"
+    - jdk: oraclejdk8
+      env: PYSPARK_PYTHON=python3 SPARK_VERSION=2.3.0 SPARK_BUILD="spark-${SPARK_VERSION}-bin-hadoop2.7"
```
Comment (felixcheung) [inline .travis.yml]: can you add the python2 mix

Comment (profes) [reply .travis.yml]: done

Comment (profes) [reply .travis.yml]: can I make version 2.3 default in `build.sbt`:
`val sparkVer = sys.props.getOrElse("spark.version", "2.3.0")`?

**Comment by felixcheung** [comment]
Yes!

**Comment by profes** [comment]
done!

File: build.sbt [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
   case "2.0" => "2.11.8"
   case "2.1" => "2.11.8"
   case "2.2" => "2.11.8"
+  case "2.3" => "2.11.8"
   case _ => throw new IllegalArgumentException(s"Unsupported Spark version: $sparkVer.")
 }
 val scalaVer = sys.props.getOrElse("scala.version", defaultScalaVer)
 val defaultScalaTestVer = scalaVer match {
-  case s if s.startsWith("2.10") => "2.0"
+  case s if s.startsWith("2.10") => "2.2.6"
```
[the remark this answers is not in the document]
Comment (profes) [reply build.sbt]: I was wondering why this 2.10 switch. sorry, this should not go in this pr.
thanks for quick merge!

**Comment by felixcheung** [comment]
merged, thanks!

**Pull request merged**